JS/share.js

```diff
@@ -0,0 +1,152 @@
+    description: "<strong>11%</strong> of the world’s population is currently vulnerable to climate change impacts such as droughts, floods, heat waves, extreme weather events and sea-level rise",
+    href: "issues.html",
+  },
+
+  {
+    type: "Social",
+    type2: "Economic",
+    title: "Refugees",
+    src: "https://www.theelders.org/sites/default/files/migrated4/syrian-refugee-child-beach-600x400.jpg",
+    description: "At least <strong> 79.5 Million </strong> people around the world have been forced to flee their homes. Among them are nearly <strong>26 Million</strong> refugees, around half of whom are under the age of 18",
+    href: "issues.html",
+  },
+
+  {
+    type: "Economic",
+    type2: "Social",
+    title: "Gender Equality",
+    src: "https://i1.wp.com/www.travindy.com/wp-content/uploads/2017/12/PATA-will-highlight-the-issue-of-gender-equality-in-2018-Dec122017.png?resize=710%2C502&ssl=1",
+    description: "Despite progress, gender inequality continues to hold women and girls back and deprive them of basic rights and opportunities.",
+    href: "issues.html",
+  },
+
+  {
+    type: "Social",
+    type2: "",
+    title: "Domestic Violence",
+    src: "https://media3.s-nbcnews.com/j/newscms/2018_43/2616181/181023-domestic-violence-stock-ac-618p_5c28e9c89f761909895faa939e650813.fit-2000w.jpg",
+    description: "About <strong>1 in 3 (35%) </strong> of women worldwide have experienced either physical and/or sexual intimate partner violence or non-partner sexual violence in their lifetime.",
+    href: "issues.html",
+  },
+];
+
+
+
+function createHtml() {
+
+  let issuesHtml = '';
+  for (let i = 0; i < issues.length; i++) {
+    issuesHtml += `<div id=${issues[i].class}><img src=${issues[i].src}></img><h5>${issues[i].title}</h5> <p>${issues[i].description}</p> <button> <a href=${issues[i].href}>View Detail</a></button></div>`;
+
+
+  }
+
+  var container = document.getElementById('issues-container');
+  container.innerHTML = issuesHtml;
+
+}
+
+createHtml();
+
+
+
+var showAll = document.getElementById("show-all");
+var health = document.getElementById("health");
+var social = document.getElementById("social");
+var economic = document.getElementById("economic");
+var environment = document.getElementById("environment");
+var container = document.getElementById('issues-container');
+
+
+showAll.addEventListener("click", function () {
+  createHtml();
+});
+
+health.addEventListener("click", function () {
+  filterByType("Health");
+});
+social.addEventListener("click", function () {
+  filterByType("Social");
+
+
+});
+economic.addEventListener("click", function () {
+  filterByType("Economic");
+
+});
+environment.addEventListener("click", function () {
+  filterByType("Environment");
+});
+
+
+function filterByType(issue) {
+
+  issuesHtml = '';
+
+  let array = [];
+
+  for (let i = 0; i < issues.length; i++) {
```

Comment:
Podes fazer isto só com 1 for. Percorres os issues e se a condição for true adicionas aos issuesHtml.

Reply:
Podes até ter só uma função de createHTML que permite a passagem de um filtro de categoria e/ou um filtro de texto
```
createHtml(filterCategory, filterName) {

for(issues) {
  if((!filterCategory || categoryConditionIsTrue) && (!filterName || nameConditionIsTrue)) {
   issuesHTML += ....
  }
 }
}

filterByType(type) {
 createHtml(filterCategory);
}

filterByName(input) {
createHtml(undefined, filterName);
}
```

Deixei-te este pseudocodigo para te ajudar. Diz-me pff se percebes :)